docs/contributing.rst

```diff
@@ -62,26 +62,176 @@ The following Python implementations are supported:
+a great place to start instrumenting. There are a couple of considerations when
+dealing with exceptions in ``ddtrace``:
+
+    - Re-raising the exception: it is crucial that we do not interfere with the
+      application, so exceptions must be re-raised. See the `bottle exception handling <https://github.com/DataDog/dd-trace-py/blob/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/bottle/trace.py#L50-L69>`_
+      instrumentation for an example.
+
+    - Gathering relevant information: ``ddtrace`` provides a helper for pulling
+      out this information and adding it to a span.  See the `cassandra
+      exception handling
+      <https://github.com/DataDog/dd-trace-py/blob/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/cassandra/session.py#L117-L122>`_
+      instrumentation for an example.
+
+
+Distributed Tracing
+~~~~~~~~~~~~~~~~~~~
+
+Some integrations can propagate a trace across application boundaries to other
+applications where the trace is continued. Datadog and ``ddtrace`` provide
+support for continuing a trace in another application.
+
+    - Propagating the trace example: `requests <https://github.com/DataDog/dd-trace-py/blob/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/requests/connection.py#L85-L88>`_
+    - Receiving a propagated trace example: `Django <https://github.com/DataDog/dd-trace-py/blob/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/django/middleware.py#L116-L121>`_
+
+
+Web frameworks
+++++++++++++++
+
+
+A web framework integration must do the following if possible:
+
+    - Install the WSGI or ASGI trace middlewares already provided by ``ddtrace``.
+    - Trace the duration of the request.
+    - Assign a resource name for a route.
+    - Use ``trace_utils.set_http_meta`` to set the standard http tags.
+    - Have an :ref:`internal service` name.
+    - Support distributed tracing (configurable).
+    - Provide insight to middlewares and views.
+    - Use the `SpanTypes.WEB` span type.
+
+Some example web framework integrations::
+    - `flask <https://github.com/DataDog/dd-trace-py/tree/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/flask>`_
+    - `django <https://github.com/DataDog/dd-trace-py/tree/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/django>`_
+
+
+Database libraries
+++++++++++++++++++
+
+``ddtrace`` already provides base instrumentation for the Python database API
+(PEP 249) which most database client libraries implement in the
+`ddtrace.contrib.dbapi <https://github.com/DataDog/dd-trace-py/blob/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/dbapi/__init__.py>`_
+module.
+
+Check out some of our existing database integrations for how to use the `dbapi`:
+
+    - `psycopg <https://github.com/DataDog/dd-trace-py/tree/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/psycopg>`_
+    - `mysql <https://github.com/DataDog/dd-trace-py/tree/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/mysql>`_
```

Comment:
Soon you can put Mariadb here instead ;)

Reply:
that's actually a good idea.... the docs and implementation are much better than mysql.... get it merged @ZStriker19 and we'll make it a reality!